summingbird-storm/src/main/scala/com/twitter/summingbird/storm/SpoutProvider.scala

```diff
@@ -0,0 +1,162 @@
+    node.members.collect { case Source(SpoutSource(s, parOpt)) => (s, parOpt) }.head
+  }
+
+  private def computeSpout(spout: Spout[(Timestamp, Any)]): Spout[(Timestamp, Any)] = {
+    val nodeName = stormDag.getNodeName(node)
+    node.members.reverse.foldLeft(spout.asInstanceOf[Spout[(Timestamp, Any)]]) { (spout, p) =>
+      p match {
+        case Source(_) => spout // The source is still in the members list so drop it
+        case OptionMappedProducer(_, op) =>
+          val boxed = Externalizer(op)
+          spout.flatMap { case (time, t) => boxed.get(t).map { x => (time, x) } }
+        case FlatMappedProducer(_, op) => {
+          val lockedOp = Externalizer(op)
+          spout.flatMap { case (time, t) => lockedOp.get.apply(t).map { x => (time, x) } }
+        }
+        case NamedProducer(_, _) => spout
+        case IdentityKeyedProducer(_) => spout
+        case AlsoProducer(_, _) => spout
+        case _ => sys.error("not possible, given the above call to span.\n" + p)
+      }
+    }
+  }
+
+  /**
+   * This method checks that sourceParallelism is greater than flatMapParallelism when FMMergeableWithSource is opted-in
+   */
+  private def validateParallelisms(stormDag: Dag[Storm], node: StormNode, sourceParallelism: Int): Unit = {
+    val isMergeableWithSource = getOrElse(stormDag, node, Constants.DEFAULT_FM_MERGEABLE_WITH_SOURCE).get
+    if (isMergeableWithSource) {
+      val flatMapParallelism = getOrElse(stormDag, node, Constants.DEFAULT_FM_PARALLELISM).parHint
+      require(flatMapParallelism <= sourceParallelism, s"SourceParallelism ($sourceParallelism) must be at least as high as FlatMapParallelism ($flatMapParallelism) when merging flatMap with Source")
+    }
+  }
+
+  /*
+   * This method formats the tuples to batched format and wraps the spout into KeyValueSpout.
+   */
+  private def createSpoutToFeedSummer[K, V](
+    stormDag: Dag[Storm],
+    node: StormNode,
+    builder: SummerBuilder,
+    spout: Spout[(Timestamp, (K, V))],
+    fn: Externalizer[(TopologyContext) => Unit],
+    flushExecTimeCounter: Incrementor,
+    executeTimeCounter: Incrementor): IRichSpout = {
+
+    val summerParalellism = getOrElse(stormDag, node, Constants.DEFAULT_SUMMER_PARALLELISM)
+    val summerBatchMultiplier = getOrElse(stormDag, node, Constants.DEFAULT_SUMMER_BATCH_MULTIPLIER)
+    val keyValueShards = executor.KeyValueShards(summerParalellism.parHint * summerBatchMultiplier.get)
+    val summerProducer = node.members.collect { case s: Summer[_, _, _] => s }.head.asInstanceOf[Summer[Storm, K, V]]
+    val batcher = summerProducer.store.mergeableBatcher
+    val formattedSummerSpout = spout.map {
+      case (time, (k, v)) => ((k, batcher.batchOf(time)), (time, v))
+    }
+    implicit val valueMonoid: Semigroup[V] = summerProducer.semigroup
+    new KeyValueSpout[(K, BatchID), (Timestamp, V)](getStormSpout(formattedSummerSpout, fn), builder, keyValueShards, flushExecTimeCounter, executeTimeCounter)
```

Comment:
nit: move getStormSpout(formattedSummerSpout, fn) to a separate temporary variable to reduce line length.